src/main/java/net/minecraftforge/common/crafting/IRecipeContainer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.crafting;
+
+import net.minecraft.inventory.InventoryCraftResult;
+import net.minecraft.inventory.InventoryCrafting;
+
+/**
+ * This interface is to be implemented on Container objects.
+ * For GUIs with recipe books, this allows their containers to have
+ * recipe completion and ghost recipes in their craft matrices.
+ */
+public interface IRecipeContainer {
+    public InventoryCraftResult getCraftResult();
```

Comment:
It might be good to add javadocs for these methods that point to the vanilla equivalents, so modders have a good idea of how they are implemented.

Reply:
You don't need `public` modifier too.

Reply:
I used public because other interfaces in forge did. I'll do the javadoc, though, and the braces.

Reply:
Done